src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/taskinitiation/PostTaskInitiateByIdControllerTest.java

```diff
@@ -0,0 +1,760 @@
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.springframework.http.MediaType.APPLICATION_PROBLEM_JSON_VALUE;
+import static uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.InitiateTaskOperation.INITIATION;
+import static uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskAttributeDefinition.TASK_CASE_ID;
+import static uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskAttributeDefinition.TASK_CREATED;
+import static uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskAttributeDefinition.TASK_DUE_DATE;
+import static uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskAttributeDefinition.TASK_NAME;
+import static uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskAttributeDefinition.TASK_TITLE;
+import static uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskAttributeDefinition.TASK_TYPE;
+
+public class PostTaskInitiateByIdControllerTest extends SpringBootFunctionalBaseTest {
+
+    private static final String ENDPOINT_BEING_TESTED = "task/{task-id}";
+
+    private Headers authenticationHeaders;
+
+    private static final Stream<String> caseManagerAndTribunalCaseworkerTasks =
+        Stream.of("reviewRespondentEvidence", "followUpOverdueRespondentEvidence",
+                  "reviewAppealSkeletonArgument", "followUpOverdueCaseBuilding", "reviewReasonsForAppeal",
+                  "followUpOverdueReasonsForAppeal", "reviewClarifyingQuestionsAnswers",
+                  "followUpOverdueClarifyingAnswers", "reviewRespondentResponse",
+                  "followUpOverdueRespondentReview", "reviewHearingRequirements",
+                  "followUpOverdueHearingRequirements", "reviewCmaRequirements",
+                  "reviewAdditionalHomeOfficeEvidence", "reviewAdditionalAppellantEvidence",
+                  "reviewAdditionalHomeOfficeEvidence", "reviewAdditionalAppellantEvidence",
+                  "createHearingBundle"
+        );
+
+    private static final Stream<String> nationalBusinessCentreTasks
+        = Stream.of("arrangeOfflinePayment", "markCaseAsPaid", "addListingDate");
+
+    private static final Stream<String> hearingCentreTasks
+        = Stream.of("allocateHearingJudge", "uploadHearingRecording");
+
+    private static final Stream<String> judgeTasks =
+        Stream.of("reviewHearingBundle", "generateDraftDecisionAndReasons", "uploadDecision",
+                  "reviewAddendumHomeOfficeEvidence", "reviewAddendumAppellantEvidence", "reviewAddendumEvidence"
+        );
+
+    @Before
+    public void setUp() {
+        //Reset role assignments
+        authenticationHeaders = authorizationHeadersProvider.getTribunalCaseworkerAAuthorization("wa-ft-test-r2");
+        common.clearAllRoleAssignments(authenticationHeaders);
```

Comment:
```suggestion

```

Minor suggestion no need to clear the role-assignments since we use one user per test, so it reduces the need to call role-assignment delete endpoints which can be a bit unreliable